internal/internal_worker_base.go

```diff
@@ -304,10 +307,13 @@ func (bw *baseWorker) pollTask() {
 	bw.retrier.Throttle(bw.stopCh)
 	if bw.pollLimiter == nil || bw.pollLimiter.Wait(bw.limiterContext) == nil {
 		task, err = bw.options.taskWorker.PollTask()
-		if err != nil && enableVerboseLogging {
-			bw.logger.Debug("Failed to poll for task.", tagError, err)
-		}
 		if err != nil {
+			// Log the error as warn if it doesn't match the last error seen
+			if err.Error() != bw.lastPollTaskErrMessage {
+				bw.logger.Warn("Failed to poll for task.", tagError, err)
+				bw.lastPollTaskErrMessage = err.Error()
```

Comment:
Just want to verify that we're definitely single-threaded through here?

Reply:
It might make sense to log every few attempts to increase visibility otherwise it might seem like the process is stuck.

Reply:
> Just want to verify that we're definitely single-threaded through here?

A good question and definitely yes. It was the reason for the comment:

> Only used during in pollTask to keep from spamming logs

Though I should have said "in pollTask non-concurrently"

Reply:
> It might make sense to log every few attempts to increase visibility otherwise it might seem like the process is stuck.

Not sure such a message counter mechanism is that warranted but if you think it is needed I can add it

Reply:
I would be a bit more noisy here to notify users that the SDK is still retrying

Reply:
Ok, I will add the counter mechanism. I am not a fan of noise or extra code/logic for these conditions, but I don't have a strong enough opinion.

Reply:
I added a lock (turns out it was needed, sorry @mmcshane) and suppressed duplicate error logs for a certain period